client/allocrunner/taskrunner/task_runner.go

```diff
@@ -510,6 +510,19 @@ func (tr *TaskRunner) initLabels() {
 		},
 	}
 
+	if tr.clientConfig.IncludeMetadataInMetrics {
+		for meta := range tr.task.Meta {
+			if len(tr.clientConfig.AllowedMetadataKeysInMetrics) > 0 && !slices.Contains(tr.clientConfig.AllowedMetadataKeysInMetrics, meta) {
+				continue
+			}
+
+			tr.baseLabels = append(tr.baseLabels, metrics.Label{
+				Name:  strings.ReplaceAll(meta, "-", "_"),
+				Value: tr.alloc.Job.Meta[meta],
```

Comment:
If we do `for meta, metaValue := range tr.task.Meta` above, we'd have the value already I think?